src/aiida/engine/processes/calcjobs/tasks.py

```diff
@@ -567,7 +575,11 @@ async def execute(self) -> plumpy.process_states.State:  # type: ignore[override
         except TransportTaskException as exception:
             raise plumpy.process_states.PauseInterruption(f'Pausing after failed transport task: {exception}')
         except plumpy.process_states.KillInterruption as exception:
-            await self._kill_job(node, transport_queue)
+            await self._kill_job(node, transport_queue, force_kill=False)
+            node.set_process_status(str(exception))
+            return self.retrieve(monitor_result=self._monitor_result)
+        except plumpy.process_states.ForceKillInterruption as exception:
+            await self._kill_job(node, transport_queue, force_kill=True)
```

Comment:
This is a bit of a roundabout way I think. If you catch the `ForceKillInterruption` why are you now calling `self._kill_job` with `force_kill=True`. That method will not do anything but log a message and straight away raise `ForceKillInterruption` again. Can't you simply get rid of the `force_kill` argument to `_kill_job` and `task_kill_job` and here simply log the message that the calcjob is force killed and you immediately go to the retrieve step?

Reply:
This was a very good suggestion, somehow I got blinded by tracing many functions..
Now the logging is done immediately following the exception.